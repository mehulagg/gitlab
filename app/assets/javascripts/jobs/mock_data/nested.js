export default {
  lines: [
    {
      offset: 0,
      content: [{ text: 'Hello' }],
    },
    {
      offset: 5,
      content: [{ text: 'foo' }],
      sections: ['prepare-script'],
      section_header: true,
    },
    {
      offset: 52,
      content: [{ text: 'bar' }],
      sections: ['prepare-script', 'prepare-script-nested'],
      section_header: true,
      section_duration: "00:02",
    },
    {
      offset: 80,
      content: [{ text: 'this is a collapsible nested section' }],
      sections: ['prepare-script', 'prepare-script-nested'],
    },
    {
      offset: 106,
      content: [],
      sections: ['prepare-script'],
      section_duration: "00:03",
    },
    {
      offset: 155,
      content: [],
    },
  ],
};
