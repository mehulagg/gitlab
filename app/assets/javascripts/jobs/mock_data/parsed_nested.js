export default {
  lines: [
    {
      offset: 0,
      content: [{ text: 'Hello' }],
      lineNumber: 0,
    },
    {
      offset: 5,
      section_header: true,
      isHeader: true,
      isClosed: true,
      line: {
        content: [{ text: 'foo' }],
        sections: ['prepare-script'],
        lineNumber: 1,
      },
      section_duration: "00:03",
      lines: [
        {
          section_header: true,
          section_duration: "00:02",
          isHeader: true,
          isClosed: true,
          line: {
            offset: 52,
            content: [{ text: 'bar' }],
            sections: ['prepare-script', 'prepare-script-nested'],
            lineNumber: 2,
          },
          lines: [
            {
              offset: 80,
              content: [{ text: 'this is a collapsible nested section' }],
              sections: ['prepare-script', 'prepare-script-nested'],
              lineNumber: 3.
            }
          ]
        }
      ]
    },
  ],
};
