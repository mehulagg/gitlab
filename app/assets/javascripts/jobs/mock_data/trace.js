// Using https://gitlab.com/gitlab-org/gitlab-ce/-/jobs/278912562

export default {
  id: 278912562,
  status: 'failed',
  complete: true,
  state:
    'eyJvZmZzZXQiOjQ5NzU5NCwibl9vcGVuX3RhZ3MiOjAsImZnX2NvbG9yIjpudWxsLCJiZ19jb2xvciI6bnVsbCwic3R5bGVfbWFzayI6MCwic2VjdGlvbnMiOltdLCJsaW5lbm9faW5fc2VjdGlvbiI6MTJ9',
  append: false,
  truncated: false,
  offset: 0,
  size: 497594,
  total: 497594,

  lines: [
    // {
    //   offset: 1000,
    //   content: [{ text: 'Running with gitlab-runner 12.1.0 (de7731dd)' }],
    // },
    // {
    //   offset: 1001,
    //   content: [{ text: ' on docker-auto-scale-com 8a6210b8' }],
    // },
    // {
    //   offset: 1002,
    //   content: [
    //     {
    //       text:
    //         'Using Docker executor with image dev.gitlab.org:5005/gitlab/gitlab-build-images:ruby-2.6.3-golang-1.11-git-2.22-chrome-73.0-node-12.x-yarn-1.16-postgresql-9.6-graphicsmagick-1.3.33',
    //     },
    //   ],
    //   section: 'prepare-executor',
    //   section_header: true,
    // },
    // {
    //   offset: 1003,
    //   content: [{ text: 'Starting service postgres:9.6.14 ...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1004,
    //   content: [{ text: 'Pulling docker image postgres:9.6.14 ...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1005,
    //   content: [
    //     {
    //       text: `Using docker image sha256:ac400042d32f8e146477ae5b3528f1e4232fc8e83b2fd9fe0cbe08900e45bae1 for
    //   postgres:9.6.14 ...`,
    //     },
    //   ],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1006,
    //   content: [{ text: 'Starting service redis:alpine ...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1007,
    //   content: [{ text: 'Pulling docker image redis:alpine ...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1008,
    //   content: [
    //     {
    //       text: `Using docker image sha256:ed7d2ff5a6232b43bdc89a2220ed989f532c3794422aa2a86823b8bc62e71447 for
    //   redis:alpine ...`,
    //     },
    //   ],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1009,
    //   content: [{ text: 'Waiting for services to be up and running...' }],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1010,
    //   content: [
    //     {
    //       text: `Pulling docker image
    //   dev.gitlab.org:5005/gitlab/gitlab-build-images:ruby-2.6.3-golang-1.11-git-2.22-chrome-73.0-node-12.x-yarn-1.16-postgresql-9.6-graphicsmagick-1.3.33
    //   ...`,
    //     },
    //   ],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1011,
    //   content: [
    //     {
    //       text: `Using docker image sha256:f5b71ccc7a53408ea88959433baf23f7d716bb431d0475a4f3d9ba213d7841c1 for
    //   dev.gitlab.org:5005/gitlab/gitlab-build-images:ruby-2.6.3-golang-1.11-git-2.22-chrome-73.0-node-12.x-yarn-1.16-postgresql-9.6-graphicsmagick-1.3.33
    //   ...`,
    //     },
    //   ],
    //   section: 'prepare-executor',
    // },
    // {
    //   offset: 1012,
    //   content: [],
    //   section: 'prepare-executor',
    //   section_footer: true,
    //   section_duration: '10:00'
    // },
    // {
    //   offset: 1013,
    //   content: [
    //     {
    //       text: `Running on runner-8a6210b8-project-13083-concurrent-0 via
    //   runner-8a6210b8-gsrm-1566561403-37c881a0...`,
    //     },
    //   ],
    //   section: 'prepare-script',
    //   section_header: true,
    // },

    // {
    //   offset: 1014,
    //   content: [
    //     { text: 'Fetching changes with git depth set to 20..', style: 'term-fg-l-green term-bold' },
    //   ],
    //   section: 'get-sources',
    //   section_header: true,
    // },
    // {
    //   offset: 1015,
    //   content: [
    //     {
    //       text: 'Reinitialized existing Git repository in /builds/gitlab-org/gitlab-ce/.git/ ',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1016,
    //   content: [
    //     {
    //       text: 'From https://gitlab.com/gitlab-org/gitlab-ce',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1017,
    //   content: [
    //     {
    //       text: ' * [new branch] 66454-create-feature-flag -&gt; origin/66454-create-feature-flag',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1018,
    //   content: [
    //     {
    //       text: 'Checking out 061606b4 as 66454-create-feature-flag...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1019,
    //   content: [
    //     {
    //       text: 'Removing .gitlab_shell_secret',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1020,
    //   content: [
    //     {
    //       text: 'Removing .gitlab_workhorse_secret',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1021,
    //   content: [
    //     {
    //       text: 'Removing config/database_geo.yml',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1022,
    //   content: [
    //     {
    //       text: 'Removing config/gitlab.yml',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1023,
    //   content: [
    //     {
    //       text: 'Removing config/redis.cache.yml',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1024,
    //   content: [
    //     {
    //       text: 'Skipping Git submodules setup',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1025,
    //   content: [],
    //   section: 'get-sources',
    // },
    // {
    //   offset: 1026,
    //   content: [],
    //   section: 'get-sources',
    //   section_footer: true,
    //   section_duration: '30:00'
    // },
    // {
    //   offset: 1027,
    //   content: [
    //     {
    //       text: 'Checking cache for debian-stretch-ruby-2.6.3-node-12.x-3...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   sections: ['restore-cache'],
    //   section_header: true,
    // },
    // {
    //   offset: 1028,
    //   content: [],
    //   sections: ['restore-cache'],
    // },
    // {
    //   offset: 1029,
    //   content: [
    //     {
    //       text:
    //         'Downloading cache.zip from https://storage.googleapis.com/gitlab-com-runners-cache/project/13083/debian-stretch-ruby-2.6.3-node-12.x-3',
    //     },
    //   ],
    //   sections: ['restore-cache'],
    // },
    // {
    //   offset: 1030,
    //   content: [],
    //   sections: ['restore-cache'],
    // },
    // {
    //   offset: 1031,
    //   content: [
    //     {
    //       text: 'Successfully extracted cache',
    //       style: 'term-fg-l-green term-bold ',
    //     },
    //   ],
    //   sections: ['restore-cache'],
    // },
    // {
    //   offset: 1032,
    //   content: [],
    //   sections: ['restore-cache'],
    //   section_footer: true,
    //   section_duration: '00:10'
    // },
    // {
    //   offset: 1033,
    //   content: [
    //     {
    //       text: 'Downloading artifacts for compile-assets pull-cache (278912482)...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    //   section_header: true,
    // },
    // {
    //   offset: 1034,
    //   content: [],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1035,
    //   content: [
    //     {
    //       text: 'Downloading artifacts from coordinator... ok ',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //     {
    //       text: ' id',
    //     },
    //     {
    //       text: '=278912482 responseStatus',
    //     },
    //     {
    //       text: '=200 OK token',
    //     },
    //     {
    //       text: '=DUNW1asp',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1036,
    //   content: [
    //     {
    //       text: 'Downloading artifacts for setup-test-env (278912484)...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1037,
    //   content: [
    //     {
    //       text: 'Downloading artifacts from coordinator... ok  id=278912484 responseStatus=200 OK token=gdMnE5zT',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1038,
    //   content: [
    //     {
    //       text: `WARNING: tmp/tests/gitlab-shell/.gitlab_shell_secret: chmod
    //       tmp/tests/gitlab-shell/.gitlab_shell_secret: no such file or directory (suppressing
    //       repeats)`,
    //       style: 'term-fg-yellow',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1039,
    //   content: [
    //     {
    //       text: 'Downloading artifacts for retrieve-tests-metadata (278912485)...',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1040,
    //   content: [
    //     {
    //       text: 'Downloading artifacts from coordinator... ok id=278912485 responseStatus=200 OK token=rCtxUhsy',
    //     },
    //   ],
    //   sections: ['download-artifacts'],
    // },
    // {
    //   offset: 1041,
    //   content: [],
    //   sections: ['download-artifacts'],
    //   section_footer: true,
    //   section_duration: '01:00:58'
    // },
    // {
    //   offset: 1042,
    //   content: [
    //     {
    //       text: 'Downloading artifacts from coordinator... ok ',
    //     },
    //     {
    //       text: ' id',
    //       style: 'term-fg-l-green term-bold',
    //     },
    //     {
    //       text: '=278912484 responseStatus=200 OK token=gdMnE5zT',
    //     },
    //   ],
    //   sections: 'download-artifacts',
    // },
    {
      offset: 1043,
      content: [{ text: 'Hello' }],
    },
    {
      offset: 5,
      content: [{ text: 'foo' }],
      section: 'prepare-script',
      section_header: true,
    },
    {
      offset: 52,
      content: [{ text: 'bar' }],
      section: 'prepare-script-nested',
      section_header: true,
    },
    {
      offset: 80,
      content: [{ text: 'this is a collapsible nested section' }],
      section: 'prepare-script-nested',
    },
    {
      offset: 90,
      section: 'prepare-script-nested',
      section_duration: '00:02',
    },
    {
      offset: 106,
      content: [],
      section: 'prepare-script',
      section_duration: '00:03',
    },
    {
      offset: 107,
      content: [],
      section_duration: '00:04',
    },
    {
      offset: 155,
      content: [],
    },
  ],
};
