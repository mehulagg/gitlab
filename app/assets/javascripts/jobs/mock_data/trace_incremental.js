// Using https://gitlab.com/gitlab-org/gitlab-ce/-/jobs/278912562

export default {
  id: 278912562,
  status: 'failed',
  complete: true,
  state:
    'eyJvZmZzZXQiOjQ5NzU5NCwibl9vcGVuX3RhZ3MiOjAsImZnX2NvbG9yIjpudWxsLCJiZ19jb2xvciI6bnVsbCwic3R5bGVfbWFzayI6MCwic2VjdGlvbnMiOltdLCJsaW5lbm9faW5fc2VjdGlvbiI6MTJ9',
  append: false,
  truncated: false,
  offset: 0,
  size: 497594,
  total: 497594,

  lines: [
    {
      offset: 1042,
      content: [
        {
          text:'Not downloading anything ',
        },
        {
          text: ' id',
          style: 'term-fg-l-green term-bold',
        },
        {
          text: '=278912484 responseStatus=200 OK token=gdMnE5zT',
        },
      ],
      sections: ['download-artifacts'],
    },

    {
      offset: 1043,
      content: [
        {
          text: 'A filipa Ã© fixe e o resto que se lixe ',
        },
        {
          text: ' id',
          style: 'term-fg-l-green term-bold',
        },
        {
          text: '=278912484 responseStatus=200 OK token=gdMnE5zT',
        },
      ],
    },
  ],
};
