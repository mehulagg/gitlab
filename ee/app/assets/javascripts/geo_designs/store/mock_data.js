import { __ } from '~/locale';

function generateRandomDateString() {
  return new Date(
    new Date(2019, 0, 1).getTime() +
      Math.random() * (new Date().getTime() - new Date(2019, 0, 1).getTime()),
  ).toString();
}

const mockData = [
  {
    id: 1,
    name: __("Zack's Design Repo"),
    url: 'http://localhost:3002',
    sync_status: 'synced',
    last_synced_at: generateRandomDateString(),
    last_verified_at: generateRandomDateString(),
    last_checked_at: generateRandomDateString(),
  },
  {
    id: 2,
    name: __("Valery's Design Repo"),
    url: 'http://localhost:3002',
    sync_status: 'pending',
    last_synced_at: generateRandomDateString(),
    last_verified_at: generateRandomDateString(),
    last_checked_at: generateRandomDateString(),
  },
  {
    id: 3,
    name: __("Mike's Design Repo"),
    url: 'http://localhost:3002',
    sync_status: 'failed',
    last_synced_at: generateRandomDateString(),
    last_verified_at: generateRandomDateString(),
    last_checked_at: generateRandomDateString(),
  },
  {
    id: 4,
    name: __("Rachel's Design Repo"),
    url: 'http://localhost:3002',
    sync_status: null,
    last_synced_at: null,
    last_verified_at: null,
    last_checked_at: null,
  },
];

export default mockData;
