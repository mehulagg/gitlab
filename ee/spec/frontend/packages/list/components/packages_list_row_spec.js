import Vuex from 'vuex';
import { mount, shallowMount, createLocalVue } from '@vue/test-utils';
import PackagesListRow from 'ee/packages/list/components/packages_list_row.vue';
import PackageTags from 'ee/packages/shared/components/package_tags.vue';
import { packageList } from '../../mock_data';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('packages_list_row', () => {
  let wrapper;
  let store;

  const [packageWithoutTagsOrPipeline, packageWithTabsAndPipeline] = packageList;

  const findPackageTags = () => wrapper.find(PackageTags);
  const findProjectLink = () => wrapper.find({ ref: 'packages-row-project' });
  const findDeleteButton = () => wrapper.find({ ref: 'action-delete' });
  const findPipelineRef = () => wrapper.find({ ref: 'pipeline-ref' });
  const findPipelineSha = () => wrapper.find({ ref: 'pipeline-sha' });

  const mountComponent = (
    isGroupPage = false,
    packageEntity = packageWithoutTagsOrPipeline,
    shallow = true,
  ) => {
    const mountFunc = shallow ? shallowMount : mount;

    const state = {
      config: {
        isGroupPage,
      },
    };

    store = new Vuex.Store({
      state,
      getters: {
        getCommitLink: () => () => {
          return 'commit-link';
        },
      },
    });

    wrapper = mountFunc(PackagesListRow, {
      localVue,
      store,
      propsData: {
        packageEntity,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  it('renders', () => {
    mountComponent();
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('tags', () => {
    it('renders package tags when a package has tags', () => {
      mountComponent(false, packageWithTabsAndPipeline);

      expect(findPackageTags().exists()).toBe(true);
    });

    it('does not render when there are no tags', () => {
      mountComponent();

      expect(findPackageTags().exists()).toBe(false);
    });
  });

  describe('when is isGroupPage', () => {
    beforeEach(() => {
      mountComponent(true);
    });

    it('has project field', () => {
      expect(findProjectLink().exists()).toBe(true);
    });

    it('does not show the delete button', () => {
      expect(findDeleteButton().exists()).toBe(false);
    });
  });

  describe('pipeline information', () => {
    it('displays branch and commit when pipeline info exists', () => {
      mountComponent(false, packageWithTabsAndPipeline);

      expect(findPipelineRef().exists()).toBe(true);
      expect(findPipelineSha().exists()).toBe(true);
    });

    it('does not show any pipeline details when no information exists', () => {
      mountComponent(false, packageWithoutTagsOrPipeline);

      expect(findPipelineRef().exists()).toBe(false);
      expect(findPipelineSha().exists()).toBe(false);
    });
  });

  describe('delete event', () => {
    beforeEach(() => mountComponent(false, packageWithoutTagsOrPipeline, false));

    it('emits the packageToDelete event when the delete button is clicked', () => {
      findDeleteButton().trigger('click');

      return wrapper.vm.$nextTick().then(() => {
        expect(wrapper.emitted('packageToDelete')).toBeTruthy();
        expect(wrapper.emitted('packageToDelete')[0]).toEqual([packageWithoutTagsOrPipeline]);
      });
    });
  });
});
