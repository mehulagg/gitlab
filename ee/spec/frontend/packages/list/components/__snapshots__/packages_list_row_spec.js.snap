// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`packages_list_row renders 1`] = `
<div
  class="gl-responsive-table-row"
  data-qa-selector="packages-row"
>
  <div
    class="table-section section-50 d-flex flex-md-column justify-content-between flex-wrap"
  >
    <div
      class="d-flex align-items-center mr-2"
    >
      <gl-link-stub
        class="text-dark font-weight-bold mb-md-1"
        href="foo"
      >
        
        Test package
      
      </gl-link-stub>
       
      <!---->
    </div>
     
    <div
      class="d-flex text-secondary text-truncate mt-md-2"
    >
      <span>
        1.0.0
      </span>
       
      <!---->
       
      <div
        class="d-flex align-items-center"
      >
        <gl-icon-stub
          class="text-secondary ml-2 mr-1"
          name="review-list"
          size="16"
        />
         
        <gl-link-stub
          class="text-secondary"
          href="/foo/bar/baz"
        >
          
        </gl-link-stub>
      </div>
       
      <div
        class="d-flex align-items-center"
      >
        <gl-icon-stub
          class="text-secondary ml-2 mr-1"
          name="package"
          size="16"
        />
         
        <span>
          Maven
        </span>
      </div>
    </div>
  </div>
   
  <div
    class="table-section d-flex flex-md-column justify-content-between align-items-md-end section-40"
  >
    <publish-method-stub
      packageentity="[object Object]"
    />
     
    <div
      class="text-secondary order-0 order-md-1 mt-md-2"
    >
      <gl-sprintf-stub
        message="Created %{timestamp}"
      />
    </div>
  </div>
   
  <div
    class="table-section section-10 d-flex justify-content-end"
  >
    <gl-new-button-stub
      aria-label="Remove package"
      category="primary"
      icon="remove"
      size="medium"
      title="Remove package"
      variant="danger"
    />
  </div>
</div>
`;
